apps/passport-client/components/modals/AccountImportModal.tsx

```diff
@@ -0,0 +1,253 @@
+    onFilesSelected: () => {
+      setImportState({ state: "ready" });
+    }
+  });
+
+  const modal = useModal();
+  const pcdCollection = usePCDCollection();
+  const dispatch = useDispatch();
+
+  const [importState, setImportState] = useState<ImportState>({
+    state: "ready"
+  });
+
+  // Called when a valid file has been selected, and the user chooses to import
+  // PCDs from it.
+  const importPCDs = useCallback(() => {
+    // Should never happen, but makes TypeScript happy that we checked for it
+    if (importState.state !== "valid-file-selected") return;
+
+    dispatch({
+      type: "merge-import",
+      collection: importState.collection,
+      pcdsToMergeIds: importState.pcdsToMergeIds
+    });
+
+    setImportState({ state: "import-complete" });
+  }, [dispatch, importState]);
+
+  // Responds to the user having selected a file to import, or to changes in
+  // the user's current PCD collection.
+  useEffect(() => {
+    (async () => {
+      // If a file has been selected, and isn't invalid or already imported
+      if (
+        filesContent.length > 0 &&
+        importState.state !== "import-complete" &&
+        importState.state !== "invalid-file"
+      ) {
+        let importedCollection: PCDCollection;
+
+        // If the file hasn't been processed yet, process it
+        if (importState.state === "ready") {
+          try {
+            // Parse the file content as JSON
+            const storageExport = JSON.parse(filesContent[0].content);
+            // Deserialize the storage - throws an error if the content is not
+            // recognized
+            const importedBackup = await deserializeStorage(
+              storageExport,
+              await getPackages()
+            );
+            importedCollection = importedBackup.pcds;
+            if (!(importedCollection instanceof PCDCollection)) {
+              throw new Error("Did not deserialize a valid PCD collection");
+            }
+          } catch (e) {
+            // The file is not valid, so bail out
+            setImportState({ state: "invalid-file" });
+            return;
+          }
+        } else {
+          importedCollection = importState.collection;
+        }
+
+        const userHasSemaphoreIdentity =
+          pcdCollection.getPCDsByType(SemaphoreGroupPCDTypeName).length > 0;
+
+        const userHasEmailPCD =
+          pcdCollection.getPCDsByType(EmailPCDTypeName).length > 0;
+
+        // Before importing, we want to filter the PCDs down to those which
+        // are valid to import, so we can tell the user how many new PCDs to
+        // expect
+        const preImportFilter = (pcd: PCD) => {
+          // If the user has a semaphore identity PCD, don't import another
+          if (
+            userHasSemaphoreIdentity &&
+            pcd.type === SemaphoreIdentityPCDTypeName
+          ) {
+            return false;
+          }
+
+          // If the user has an email PCD, don't import another
```

Comment:
I hadn't thought about this.  Seems valid for now, though I think multiple emails is a place we'll want to go in future.  Wouldn't email reissuance delete the extra one anyway?

Reply:
But by not importing it at all, we avoid the unnecessary sync after it gets deleted!